### DISCUSSION TITLE
Quickfix stop at end of list

### DISCUSSION THREAD

**Discussion by 9mm** [nvim-mini/mini.nvim#583] [Q&A] [answered]
Hey, really nice job on bracketed. Is there a way to stop at the end of `]q` similar to how "cn" works? If it wraps back around I can never tell where im at in the list, i was thinking maybe this could be an option?

**Comment by echasnovski** [comment, marked as answer]
> Hey, really nice job on bracketed. Is there a way to stop at the end of `]q` similar to how "cn" works? If it wraps back around I can never tell where im at in the list, i was thinking maybe this could be an option?

Hi!

Sure. Almost all targets support `wrap` flag which is `true` by default. To not wrap for `quickfix` target, use the following setup:
```lua
require('mini.bracketed').setup({ quickfix = { options = { wrap = false } } })
```

**Comment by 9mm** [comment]
So awesome, thank you. I read in the docs I guess I missed that. Super awesome!!

**Comment by 9mm** [comment]
Btw I already made a post praising your plugins but just want to say again, I feel like you REALLY get the ... "energy" of neovim/vim very nicely by making plugins that dont overcomplicate things, and do their purpose very well.

I don't know if you are looking for suggestions, but redoing the quickfix window would be amazing.

Right now I'm using an amalgamation of...

1. https://github.com/yorickpeterse/nvim-pqf for formatting, although I wish he would format into columns, he said he's not going to do that, but at least its better than the defualt
2. A very ugly keymap:

```
local function toggle_quickfix()
  for _, win in pairs(vim.api.nvim_list_wins()) do
    if vim.fn.getbufvar(vim.api.nvim_win_get_buf(win), "&buftype") == "quickfix" then
      vim.cmd("cclose")
      return
    end
  end
  vim.cmd("copen")
end

vim.keymap.set("n", "<Leader>qf", function() toggle_quickfix() end, { desc = "Quickfix list" })
```
3. Your mini.bracketed plugin for `[q` and `]q`

I have considered other plugins like https://github.com/stevearc/qf_helper.nvim
However, how he does the Q commands, they don't work for me 100%, they seem kind of broken, yours work great in bracketed though. The only thing I think that would be needed is a toggle feature which doesnt requrie the cursor being in the QF window, as way as maybe some way to scan past them without entering every file, etc

**Reply by echasnovski** [reply]
> Btw I already made a post praising your plugins but just want to say again, I feel like you REALLY get the ... "energy" of neovim/vim very nicely by making plugins that dont overcomplicate things, and do their purpose very well.

:heart: 

> I don't know if you are looking for suggestions, but redoing the quickfix window would be amazing.

Yes, it is [planned](https://github.com/echasnovski/mini.nvim#planned-modules). Not quite soon, though, as I would like to focus on getting essential things to fully replace important parts of my config with 'mini.nvim'.
Not sure how 'mini.quickfix' would look, but I'll think something. I heard (and experienced briefly) that quickfix/loclist functionality is somewhat hacky, so not really looking forward to working with it :sweat_smile:
